crates/biome_service/src/settings.rs

```diff
@@ -862,46 +891,59 @@ impl OverrideSettingPattern {
         if let Some(indent_width) = css_formatter.indent_width.or(formatter.indent_width) {
             options.set_indent_width(indent_width)
         }
+        if let Some(line_ending) = css_formatter.line_ending.or(formatter.line_ending) {
+            options.set_line_ending(line_ending);
+        }
         if let Some(line_width) = css_formatter.line_width.or(formatter.line_width) {
             options.set_line_width(line_width);
         }
         if let Some(quote_style) = css_formatter.quote_style {
             options.set_quote_style(quote_style);
         }
-        let mut cache = self.cached_css_format_options.write().unwrap();
-        let _ = cache.insert(options.clone());
+
+        if let Ok(mut writeonly_cache) = self.cached_css_format_options.write() {
+            let options = options.clone();
+            let _ = writeonly_cache.insert(options);
+        }
     }
 
-    fn js_parser_options(&self, options: &mut JsParserOptions) {
-        let cache = self.cached_js_parser_options.read().unwrap();
-        if let Some(cached_options) = cache.as_ref() {
-            *options = cached_options.clone();
-            return;
+    fn apply_overrides_to_js_parser_options(&self, options: &mut JsParserOptions) {
+        if let Ok(readonly_cache) = self.cached_js_parser_options.read() {
+            if let Some(cached_options) = readonly_cache.as_ref() {
+                *options = cached_options.clone();
+                return;
+            }
+            drop(readonly_cache);
```

Comment:
I *think* we can remove the `drop`, if you decide to handle the `Ok`. The reason why `drop` was there, is because `cache` didn't go out of scope, so I had to manually remove it.

But now, since you're using a `if let Ok()` block, `readonly_cache` goes out of scope in this line, automatically, which means we don't need the `drop` anymore.

To make sure that's correct, try to run the `check:apply` script that we have in the top-level `pacakge.json`. It doesn't get stuck after 3/4 times, it means the `drop` isn't needed.

Reply:
Fixed in e6adc4a.